druntime/src/core/demangle.d

```diff
@@ -210,41 +212,45 @@ pure @safe:
     }
 
 
-    void test( char val )
+    bool test( char val ) nothrow
     {
-        if ( val != front )
-            error();
+        return val == front;
     }
 
-
-    void popFront()
+    void popFront() nothrow
     {
+        import core.exception : RangeError;
+
         if ( pos++ >= buf.length )
-            error();
+            throw new RangeError;
```

Comment:
maybe you can just use an assert here instead of manually throwing error? Manually throwing errors from nothrow functions can be confusing (especially since one can catch Error - it's not recommended to catch error, but one can do it anyway.)

Reply:
Unless I'm wrong, `assert` actually throws `AssertError`. On the other hand, `RangeError` is used in such cases in the druntime code

Reply:
I don't see any cases of `RangeError` being used in druntime. We've been moving away from throwing Errors and replacing them with asserts since they don't (necessarily) depend on Throwable and can be configured (e.g. `assert(0)` can generate a HALT instruction in release builds, `-betterC` can make `assert` use C's handler etc.).

Reply:
> I don't see any cases of RangeError being used in druntime

Hm, I seen it used as base of `ArrayIndexError` but didn't dig deeper

Ok, I replace it right now